### PR TITLE
Add detailed error example and error article

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

We should document how service authors and clients should 
deal with errors and what options are available to them.

Modifications:

- Add an article explaining what the error mechanisms available
- Add an example

Result:

Better docs

### CODE REVIEW

**Comment by glbrntt** [comment]
This is split into multiple commits. Note that [b76d42f](https://github.com/grpc/grpc-swift/pull/2145/commits/b76d42ff81d1ab6a4c8dc7a90f91655c15b2d8b4) is just adding the generated code.

File: Sources/GRPCCore/Documentation.docc/Articles/Errors.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+In gRPC the outcome of every RPC is represented by a status made up of a code
+and a message. The status is propagated from the server to the client in the
+metadata as the final part of an RPC indicating the outcome of the RPC.
+
+You can find more information about the error codes in ``RPCError/Code`` and in
+the status codes guide on the
+[gRPC website](https://grpc.io/docs/guides/status-codes/).
+
+This mechanism is part of the gRPC protocol is supported by all client/server
```
Comment (gjcairo) [inline Sources/GRPCCore/Documentation.docc/Articles/Errors.md:30]:
```suggestion
This mechanism is part of the gRPC protocol and is supported by all client/server
```

File: Sources/GRPCCore/Documentation.docc/Articles/Errors.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+#### Clients
+
+Clients should catch ``RPCError`` if they are interested in the failures from an
+RPC. This is a manifestation of the error sent by the server but in some cases
+it may be synthesized locally.
+
+For clients using the rich error model, the ``RPCError`` can be caught and a
+detailed error can be extracted from it using `unpackGoogleRPCStatus()`.
+
+See [`error-details`](https://github.com/grpc/grpc-swift/tree/main/Examples) for
```
Comment (gjcairo) [inline Sources/GRPCCore/Documentation.docc/Articles/Errors.md:82]: Should we link to `Examples/error-details`?

File: Sources/GRPCCore/Documentation.docc/Articles/Errors.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+an examples.
```
Comment (gjcairo) [inline Sources/GRPCCore/Documentation.docc/Articles/Errors.md:83]:
```suggestion
an example.
```

File: Sources/GRPCCore/Documentation.docc/Articles/Errors.md (hunk 2 above)
Comment (gjcairo) [inline Sources/GRPCCore/Documentation.docc/Articles/Errors.md:82]: Should we link to `Examples/error-details`?

```suggestion
See [`error-details`](https://github.com/grpc/grpc-swift/tree/main/Examples/error-details) for
```

**Review by gjcairo — APPROVED** [review]

**Pull request merged**